### DISCUSSION TITLE
Sprint Scope

### DISCUSSION THREAD

**Discussion by gitaaron** [functionland/box#52] [General]
I’ve update our [scope](https://drive.google.com/drive/folders/1hmvAFVCClk2mD8Of6ntT6JNOhMMV0mlJ) document for this sprint.

I would like to get feedback on the following :

1.  I think we should focus on a sample web based photo gallery for this sprint and make TODO a stretch goal since it is less complimentary to Photos.

2.  I think we should consider using libp2p out of scope and instead invoke simple HTTP GET/POST using 'fetch' (or some simple alternative) for now.  Libp2p is not necessary since our scope for this sprint does not involve security, NAT traversal or other advanced P2P networking scenarios.  We may end up throwing away some code, however, I think it will be beneficial for us to gain momentum and hammer out the data modelling / reference specs.

3.  @farhoud   and I would like to raise an issue with the footprint for getting the back-end up and running.  (currently ~2 gigs)  We think the main reason for this is we are including the sample apps with the back-end.  @farhoud mentioned if we only include the back-end we can reduce the footprint to ~200 mb)  We have some proposals [here](https://www.figma.com/file/R769jdp7ZAstsqgrLPORVS/Back-end-Setup?node-id=0%3A1) on how it might work from the 'getting started' page.

**Comment by ehsan6sha** [comment]
Hi @gitaaron 
for 1, the Photos app is done and working fine even with the backend, so doing a web based app is double work
for 2, my opinion is that it deviates from the product extensively and downgrade it substantially, which might not be acceptable even as an MVP since the principles will be changed. We still require NAT traversal though, unless I am not aligned on the scope. Also we are not facing any issues here to switch to an alternative, since File and Data protocol are working fine

**Reply by gitaaron** [reply]
Hi @ehsan6sha  - perhaps I am missing something.   I thought the Photos app was not currently  working with BOX (only TODOs was).

I will try and get the Photos app up and running  with the back-end and reassess.

Thanks!

**Comment by ehsan6sha** [comment]
@gitaaron the file side is working fine. Missing features are those which depend on data protocol, such as album creation, for which we have TODO app

**Reply by gitaaron** [reply]
@ehsan6sha that is great! How about this -

1. I add a sample app to the 'examples' directory that is a simple web based photo gallery. It should not be replicating much work since I should be able to pull out the code from Photos. In the process I can document my steps and better wrap my head around how File/Data APIs actually work and have a better picture of what to put in the getting started tutorial.

2. @mehdibalouchi can take what I wrote as a template and modify it so that people can create a Todo app instead of a photo gallery (if they want).

Sound good to you?

**Reply by ehsan6sha** [reply]
yes, works! And let's use the Fula protocol and not the HTTP/fetch since the goal is to show how to create apps with Fula.

**Reply by gitaaron** [reply]
Sounds good!  Sorry for that - I am still getting my head around everything and was under the impression it was not already working yet.

**Reply by mehdibalouchi** [reply]
@gitaaron This seems good, I recommend using ReacjJS. It is simple and fast and we already have some helpers (hooks, context providers) written for react
